cmd/erasure-metadata.go

```diff
@@ -626,10 +626,16 @@ func getInternalReplicationState(m map[string]string) ReplicationState {
 	for k, v := range m {
 		switch {
 		case equals(k, ReservedMetadataPrefixLower+ReplicationTimestamp):
-			tm, _ := time.Parse(http.TimeFormat, v)
+			tm, _ := time.Parse(time.RFC3339Nano, v)
+			if tm.IsZero() {
+				tm, _ = time.Parse(http.TimeFormat, v)
+			}
```

Comment:
move this to `internal/amztime` package and call this `ParseReplicationTS(str string)` and parse using two different values return which ever succeeds.